docs/book/src/reference/ai_integration.md

```diff
@@ -0,0 +1,194 @@
+
+- Concise summary of scaffold changes
+- File list with conflict markers (if any)
+- Room for prompt engineering
+- Room for model parameters tuning.
+
+You do not get:
+
+- Automatic conflict resolution (intentionally out of scope)
+
+Required workflow permissions:
+
+- Default `GITHUB_TOKEN` with `permissions: models: read`.
+
+---
+
+### Workflow (`.github/workflows/update-ai-summary.yml`)
+
+The following workflow sets a template that can be optmized by users via prompt engineering and model parameters tuning.
+
+```yaml
+name: Alpha Update (with AI summary)
+
+permissions:
+  contents: write
+  issues: write
+  pull-requests: write
+  models: read  # Necessary permission
+
+on:
+  workflow_dispatch:
+  schedule:
+    - cron: "0 0 * * 2" # Every Tuesday at 00:00 UTC
+
+jobs:
+  alpha-update:
+    runs-on: ubuntu-latest
+    env:
+      GH_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+
+    steps:
+    - name: Checkout repository
+      uses: actions/checkout@v4
+      with:
+        token: ${{ secrets.GITHUB_TOKEN }}
+        fetch-depth: 0
+    
+    - name: Set up Go
+      uses: actions/setup-go@v5
+      with:
+        go-version: stable
+
+    - name: Configure Git
+      run: |
+        git config --local user.name "github-actions[bot]"
+        git config --local user.email "github-actions[bot]@users.noreply.github.com"
+
+    - name: Install Kubebuilder
+      run: |
+        git clone --depth 1 --branch master \
+          https://github.com/kubernetes-sigs/kubebuilder.git /tmp/kubebuilder
+        cd /tmp/kubebuilder
+        make build
+        sudo cp bin/kubebuilder /usr/local/bin/
+        kubebuilder version
+
+    - name: Run kubebuilder alpha update
+      run: |
+        kubebuilder alpha update \
+          --force \
+          --restore-path .github/workflows \
+          --push \
+          --open-gh-issue
+
+    - name: Create patch of last commit # Users may want to extract other info
+      run: |
+        git show HEAD > update.patch
+
+    - name: Prepare model prompt # Room for prompt engineering
+      run: |
+        cat > prompt.txt <<'EOF'
+        Start by stating that this summary is AI generated and can contain errors.
+        You are a concise update summarizer for an automated Kubebuilder scaffold update.
+        Analyze the patch and write a GitHub Issue comment that explains what changed.
+        Scan the patch for Git conflict markers (<<<<<<<, =======, >>>>>>>). For each conflict:
+        Explain what caused the conflict.
+        How to properly resolve them.
+        Inform the user about the recommended next steps.
+        Constraints: Markdown only, no raw diff snippets.
+
+        ---- DIFF BELOW ----
+        EOF
+        cat update.patch >> prompt.txt
+
+    - name: Generate summary with GitHub Models # Room for parameters tuning
+      env:
+        GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+      run: |
+        set -o pipefail
+        jq -Rs '
+        {
+          model: "openai/gpt-4o",
```

Comment:
That is cool so if we use this model we do not need any extra configuration permission and we will have the summary